mesonbuild/compilers/compilers.py

```diff
@@ -154,6 +154,11 @@ def is_java(fname: mesonlib.FileOrString) -> bool:
     suffix = fname.split('.')[-1]
     return suffix in lang_suffixes['java']
 
+def is_single_compile(fname: mesonlib.FileOrString) -> bool:
```

Comment:
I don't understand the name of this function. Why do Rust sources require more than one compile step?

Reply:
No it's the opposite; it's a source that can be compiled individually, as opposed to Rust which compiles by crate. The name comes from generate_single_compile in ninjabackend.py.

Reply:
maybe `is_single_tu` or `is_single_translation_unit`? I've always thought that backend name could be confusing

Reply:
I went for `is_separate_compile()`